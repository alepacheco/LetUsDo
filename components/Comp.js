function Home() {
  return <div>Imported</div>;
}

export default Home;