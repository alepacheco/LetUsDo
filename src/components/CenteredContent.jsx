import React from 'react';
import '../styles/components/CenteredContent.scss';

export const CenteredContent = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
