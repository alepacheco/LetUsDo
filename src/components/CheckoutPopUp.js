import React from 'react';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CheckoutFormWrapper from './stripe/CheckoutFormWrapper';
import { bindActionCreators } from 'redux';

import '../styles/components/CheckoutPopUp.css';
import * as actions from '../actions/checkoutPopUpActions';

const validateEmail = email => {
  // eslint-disable-next-line no-useless-escape
  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(String(email).toLowerCase());
}

class CheckoutPopUp extends React.Component {
  constructor(props) {
    super(props);

    this.onChangeEmail = this.onChangeEmail.bind(this);
  }

  onChangeEmail(event) {
    const emailValid = validateEmail(event.target.value);

    this.props.actions.setCheckoutEmailValid(emailValid);
    this.props.actions.setCheckoutEmail(event.target.value);
  }

  render() {
    return (
      <Modal
        show={this.props.show}
        onHide={this.props.onHide}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header closeButton>
          <Modal.Title id="contained-modal-title-vcenter">
            Confirm order
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div className="task-description-title">
            Task description
          </div>
          <div className="task-description-content">
            {this.props.taskText || 'No description specified'}
          </div>
          <Form>
            <Form.Group controlId="formBasicEmail" className="email-form">
              <Form.Control
                type="email"
                placeholder="Enter email"
                onChange={this.onChangeEmail}
                value={this.props.email} />
            </Form.Group>
          </Form>

          <CheckoutFormWrapper />
        </Modal.Body>

      </Modal>
    );
  }
}

const mapStateToProps = (state) => ({
  taskText: state.taskModal.taskText,
  email: state.checkoutPopUp.email
});

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CheckoutPopUp);

CheckoutPopUp.propTypes = {
  onHide: PropTypes.func,
  taskText: PropTypes.string,
  email: PropTypes.string,
  show: PropTypes.bool,
  actions: PropTypes.shape({
    setCheckoutEmailValid: PropTypes.func,
    setCheckoutEmail: PropTypes.func
  }),
};