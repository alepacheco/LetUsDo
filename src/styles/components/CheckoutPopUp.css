.email-form {
  margin-bottom: 0;
  margin-top: 16px;
}

.task-description-title {
}

.task-description-content {
  margin: 0 0 24px 0;
  color: #6b757d;
}
