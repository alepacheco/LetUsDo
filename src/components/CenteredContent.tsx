import React from 'react';
import 'src/styles/components/CenteredContent.scss';

type CenteredContentProps = {
  reverse?: boolean;
};

export const CenteredContent: React.FC<CenteredContentProps> = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
