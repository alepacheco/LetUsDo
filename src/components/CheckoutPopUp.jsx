import React from 'react';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import CheckoutFormWrapper from './stripe/CheckoutFormWrapper';
import '../styles/components/CheckoutPopUp.css';
import * as checkoutPopUpActions from '../actions/checkoutPopUpActions';
import * as taskModalActions from '../actions/taskModalActions';

const validateEmail = email => {
  // eslint-disable-next-line no-useless-escape
  const validEmailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return validEmailRegex.test(String(email).toLowerCase());
};

export class CheckoutPopUp extends React.Component {
  constructor(props) {
    super(props);

    this.onChangeEmail = this.onChangeEmail.bind(this);
    this.onHideCheckoutPopUp = this.onHideCheckoutPopUp.bind(this);
  }

  onChangeEmail(event) {
    const emailValid = validateEmail(event.target.value);

    this.props.actions.setCheckoutEmailValid(emailValid);
    this.props.actions.setCheckoutEmail(event.target.value);
  }

  onHideCheckoutPopUp() {
    this.props.actions.setDialog('closed');
  }

  render() {
    return (
      <Modal
        show={this.props.checkoutPopupState !== 'closed'}
        onHide={this.onHideCheckoutPopUp}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header closeButton>
          <Modal.Title id="contained-modal-title-vcenter">Confirm order</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div className="task-description-title">Task description</div>
          <div className="task-description-content">
            {this.props.taskText || 'No description specified'}
          </div>
          <Form>
            <Form.Group controlId="formBasicEmail" className="email-form">
              <Form.Control
                type="email"
                placeholder="Enter email"
                onChange={this.onChangeEmail}
                value={this.props.email}
              />
            </Form.Group>
          </Form>

          <CheckoutFormWrapper />
        </Modal.Body>
      </Modal>
    );
  }
}

const mapStateToProps = state => ({
  taskText: state.taskModal.taskText,
  email: state.checkoutPopUp.email,
  purchaseCompleted: state.checkoutPopUp.purchaseCompleted,
  checkoutPopupState: state.taskModal.checkoutPopupState
});

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(
      {
        ...taskModalActions,
        ...checkoutPopUpActions
      },
      dispatch
    )
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CheckoutPopUp);

CheckoutPopUp.propTypes = {
  taskText: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  checkoutPopupState: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    setCheckoutEmailValid: PropTypes.func,
    setCheckoutEmail: PropTypes.func,
    setDialog: PropTypes.func
  }).isRequired
};
