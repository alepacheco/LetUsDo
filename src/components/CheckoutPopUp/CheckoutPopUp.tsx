import React from 'react';
import Modal from 'react-bootstrap/Modal';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import '../../styles/components/CheckoutPopUp.css';
import * as checkoutPopUpActions from '../../actions/checkoutPopUpActions';
import * as taskModalActions from '../../actions/taskModalActions';
import { CheckoutContent } from './CheckoutContent';
import { PurchaseCompletedContent } from './PurchaseCompletedContent';
import { validateEmail } from '../../utils/generic';

type GetContentByTypeProps = {
  type: 'open' | 'purchaseCompleted' | 'closed';
  taskText: string;
  onChangeEmail: (email: string) => void;
  email: string;
}
const GetContentByType = ({ type, taskText, onChangeEmail, email }: GetContentByTypeProps) => {
  if (type === 'open') {
    return <CheckoutContent taskText={taskText} onChangeEmail={onChangeEmail} email={email} />;
  }
  if (type === 'purchaseCompleted') {
    return <PurchaseCompletedContent />;
  }

  return <div />;
};

type CheckoutPopUpProps = {
  actions: {
    setCheckoutEmailValid: (valid: boolean) => void;
    setCheckoutEmail: (email: string) => void;
    setDialog: (state: string) => void;
  }
  checkoutPopupState: 'open' | 'closed' | 'purchaseCompleted';
  taskText: string;
  email: string;
};
export const CheckoutPopUp: React.FC<CheckoutPopUpProps> = ({ actions, checkoutPopupState, taskText, email }) => {
  const onChangeEmail = (event: any) => {
    const emailValid = validateEmail(event.target.value);

    actions.setCheckoutEmailValid(emailValid);
    actions.setCheckoutEmail(event.target.value);
  }

  const onHideCheckoutPopUp = () => {
    actions.setDialog('closed');
  }

  return (
    <Modal
      show={checkoutPopupState !== 'closed'}
      onHide={onHideCheckoutPopUp}
      size="lg"
      aria-labelledby="contained-modal-title-vcenter"
      centered
    >
      <GetContentByType
        type={checkoutPopupState}
        taskText={taskText}
        onChangeEmail={onChangeEmail}
        email={email}
      />
    </Modal>
  );
}

const mapStateToProps = (state: any) => ({
  taskText: state.taskModal.taskText,
  email: state.checkoutPopUp.email,
  purchaseCompleted: state.checkoutPopUp.purchaseCompleted,
  checkoutPopupState: state.taskModal.checkoutPopupState
});

function mapDispatchToProps(dispatch: (action: any) => any) {
  return {
    actions: bindActionCreators(
      {
        ...taskModalActions,
        ...checkoutPopUpActions
      },
      dispatch
    )
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CheckoutPopUp);
