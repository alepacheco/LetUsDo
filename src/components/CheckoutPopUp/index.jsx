import React from 'react';
import Modal from 'react-bootstrap/Modal';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import '../../styles/components/CheckoutPopUp.css';
import * as checkoutPopUpActions from '../../actions/checkoutPopUpActions';
import * as taskModalActions from '../../actions/taskModalActions';
import { CheckoutContent } from './CheckoutContent';
import { PurchaseCompletedContent } from './PurchaseCompletedContent';
import { validateEmail } from '../../utils/generic';

const GetContentByType = ({ type, taskText, onChangeEmail, email }) => {
  if (type === 'open') {
    return <CheckoutContent taskText={taskText} onChangeEmail={onChangeEmail} email={email} />;
  }
  if (type === 'purchaseCompleted') {
    return <PurchaseCompletedContent />;
  }

  return <div />;
};

export class CheckoutPopUp extends React.Component {
  constructor(props) {
    super(props);

    this.onChangeEmail = this.onChangeEmail.bind(this);
    this.onHideCheckoutPopUp = this.onHideCheckoutPopUp.bind(this);
  }

  onChangeEmail(event) {
    const emailValid = validateEmail(event.target.value);

    this.props.actions.setCheckoutEmailValid(emailValid);
    this.props.actions.setCheckoutEmail(event.target.value);
  }

  onHideCheckoutPopUp() {
    this.props.actions.setDialog('closed');
  }

  render() {
    return (
      <Modal
        show={this.props.checkoutPopupState !== 'closed'}
        onHide={this.onHideCheckoutPopUp}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <GetContentByType
          type={this.props.checkoutPopupState}
          taskText={this.props.taskText}
          onChangeEmail={this.onChangeEmail}
          email={this.props.email}
        />
      </Modal>
    );
  }
}

const mapStateToProps = state => ({
  taskText: state.taskModal.taskText,
  email: state.checkoutPopUp.email,
  purchaseCompleted: state.checkoutPopUp.purchaseCompleted,
  checkoutPopupState: state.taskModal.checkoutPopupState
});

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(
      {
        ...taskModalActions,
        ...checkoutPopUpActions
      },
      dispatch
    )
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CheckoutPopUp);

CheckoutPopUp.propTypes = {
  taskText: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  checkoutPopupState: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    setCheckoutEmailValid: PropTypes.func,
    setCheckoutEmail: PropTypes.func,
    setDialog: PropTypes.func
  }).isRequired
};
