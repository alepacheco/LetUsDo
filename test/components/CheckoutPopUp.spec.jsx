import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CheckoutPopUp } from '../../src/components/CheckoutPopUp';

jest.mock('react-ga', () => ({ initialize: jest.fn() }));

describe('<CheckoutPopUp />', () => {
  it('renders open', () => {
    const wrapper = shallow(
      <CheckoutPopUp
        taskText="This is my task description"
        email="email@example.com"
        show
        actions={{}}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders closed', () => {
    const wrapper = shallow(
      <CheckoutPopUp
        taskText="This is my task description"
        email="email@example.com"
        show={false}
        actions={{}}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
