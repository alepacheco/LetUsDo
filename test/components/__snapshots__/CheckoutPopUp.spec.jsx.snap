// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutPopUp /> renders closed 1`] = `
<Bootstrap(Modal)
  aria-labelledby="contained-modal-title-vcenter"
  centered={true}
  onHide={[Function]}
  show={false}
  size="lg"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      id="contained-modal-title-vcenter"
    >
      Confirm order
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <div
      className="task-description-title"
    >
      Task description
    </div>
    <div
      className="task-description-content"
    >
      This is my task description
    </div>
    <Form
      inline={false}
    >
      <FormGroup
        className="email-form"
        controlId="formBasicEmail"
      >
        <FormControl
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
          value="email@example.com"
        />
      </FormGroup>
    </Form>
    <CheckoutFormWrapper />
  </ModalBody>
</Bootstrap(Modal)>
`;

exports[`<CheckoutPopUp /> renders open 1`] = `
<Bootstrap(Modal)
  aria-labelledby="contained-modal-title-vcenter"
  centered={true}
  onHide={[Function]}
  show={true}
  size="lg"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      id="contained-modal-title-vcenter"
    >
      Confirm order
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <div
      className="task-description-title"
    >
      Task description
    </div>
    <div
      className="task-description-content"
    >
      This is my task description
    </div>
    <Form
      inline={false}
    >
      <FormGroup
        className="email-form"
        controlId="formBasicEmail"
      >
        <FormControl
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
          value="email@example.com"
        />
      </FormGroup>
    </Form>
    <CheckoutFormWrapper />
  </ModalBody>
</Bootstrap(Modal)>
`;
